src/transport/TransportMgrBase.cpp

```diff
@@ -99,58 +89,33 @@ void TransportMgrBase::HandleMessageReceived(const Transport::PeerAddress & peer
 }
 
 #if INET_CONFIG_ENABLE_TCP_ENDPOINT
-void TransportMgrBase::HandleConnectionReceived(Transport::ActiveTCPConnectionState * conn)
+void TransportMgrBase::HandleConnectionReceived(Transport::ActiveTCPConnectionState & conn)
 {
     if (mSessionManager != nullptr)
     {
         mSessionManager->HandleConnectionReceived(conn);
     }
     else
     {
-        Transport::TCPBase * tcp = reinterpret_cast<Transport::TCPBase *>(conn->mEndPoint->mAppState);
-
-        // Close connection here since no upper layer is interested in the
-        // connection.
-        if (tcp)
-        {
-            tcp->TCPDisconnect(conn, /* shouldAbort = */ true);
-        }
+        // Close connection here since no upper layer is interested in the connection.
+        Transport::ActiveTCPConnectionHolder releaseConnection(&conn);
```

Comment:
It's really not obvious why this ends up closing the connection.  ActiveTCPConnectionHolder retains in its constructor... so is this relying on `conn` having a zero refcount before this line and then this going out of scope destroying it?  It seems weird to destroy something the caller is clearly referencing.  Would it not be better if this ActiveTCPConnectionHolder were in the caller, and then if no one takes a holder to the connection it would go away once that holder goes out of scope?